sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/jaas/JaasResolver.java

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.cloud.service.implementation.jaas;
+
+import com.azure.spring.cloud.service.implementation.kafka.AzureKafkaPropertiesUtils;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public final class JaasResolver {
+
+    private JaasResolver() {
+    }
+
+    public static Jaas resolve(String jaasConfig) {
+        String[] arr = extractLoginModuleAndControlFlagFromJaasString(jaasConfig);
+        Jaas jaas = new Jaas(arr[0], arr[1]);
+        jaas.setOptions(convertJaasOptionsFromStringToMap(jaasConfig));
+        return jaas;
+    }
+
+    private static String[] extractLoginModuleAndControlFlagFromJaasString(String jaasConfig) {
+        if (jaasConfig == null || !jaasConfig.contains(" ") || !jaasConfig.endsWith(";")) {
+            return new String[2];
+        }
+        return jaasConfig.substring(0, jaasConfig.length() - 1).split(" ", 3);
+    }
+
+    private static Map<String, String> convertJaasOptionsFromStringToMap(String source) {
+        if (source == null || !source.startsWith(AzureKafkaPropertiesUtils.SASL_JAAS_CONFIG_OAUTH_PREFIX) || !source.endsWith(";")) {
```

Comment:
this is another coupling, if jaas is neutral, then it shouldn't refer to anything related to Kafka.

Reply:
agree, forgot here when moving method.